File: backend/dotnet/StoreOps.Infrastructure/Entities.cs
namespace StoreOps.Infrastructure.Entities;

public sealed class CompanyEntity
{
    public Guid CompanyId { get; set; }
    public string CompanyCode { get; set; } = string.Empty;
    public string CompanyName { get; set; } = string.Empty;
    public string Status { get; set; } = "active";
}

public sealed class RegionEntity
{
    public Guid RegionId { get; set; }
    public Guid CompanyId { get; set; }
    public string RegionCode { get; set; } = string.Empty;
    public string RegionName { get; set; } = string.Empty;
}

public sealed class StoreEntity
{
    public Guid StoreId { get; set; }
    public Guid CompanyId { get; set; }
    public Guid RegionId { get; set; }
    public string StoreCode { get; set; } = string.Empty;
    public string StoreName { get; set; } = string.Empty;
    public string StoreType { get; set; } = string.Empty;
}

public sealed class EmployeeEntity
{
    public Guid EmployeeId { get; set; }
    public Guid CompanyId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public DateOnly HireDate { get; set; }
    public string EmploymentStatus { get; set; } = "active";
}

public sealed class EmployeeAssignmentEntity
{
    public Guid AssignmentId { get; set; }
    public Guid EmployeeId { get; set; }
    public Guid StoreId { get; set; }
    public Guid RegionId { get; set; }
    public Guid PositionId { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public decimal FteRatio { get; set; }
}

public sealed class ChecklistInstanceEntity
{
    public Guid ChecklistInstanceId { get; set; }
    public Guid ChecklistTemplateId { get; set; }
    public Guid StoreId { get; set; }
    public string Status { get; set; } = "planned";
    public decimal? TotalScore { get; set; }
    public decimal? ComplianceRate { get; set; }
}

public sealed class KpiActualEntity
{
    public Guid KpiActualId { get; set; }
    public Guid KpiId { get; set; }
    public string ScopeType { get; set; } = string.Empty;
    public Guid? StoreId { get; set; }
    public Guid? EmployeeId { get; set; }
    public DateOnly PeriodStart { get; set; }
    public DateOnly PeriodEnd { get; set; }
    public decimal ActualValue { get; set; }
}

public sealed class SnapshotRunEntity
{
    public Guid SnapshotRunId { get; set; }
    public DateOnly SnapshotDate { get; set; }
    public string SnapshotType { get; set; } = string.Empty;
    public DateOnly PeriodStart { get; set; }
    public DateOnly PeriodEnd { get; set; }
    public DateTimeOffset GeneratedAt { get; set; }
    public string GeneratedBy { get; set; } = string.Empty;
}

File: backend/domain/entities.ts
export type ScopeType = "company" | "region" | "store";
export type EmploymentStatus = "active" | "inactive" | "terminated" | "leave";
export type AssignmentStatus = "active" | "ended" | "planned";
export type ChecklistStatus = "planned" | "in_progress" | "completed" | "cancelled";
export type TargetDirection = "higher_is_better" | "lower_is_better";

export interface Company {
  companyId: string;
  companyCode: string;
  companyName: string;
  status: string;
  createdAt: string;
}

export interface Region {
  regionId: string;
  companyId: string;
  regionCode: string;
  regionName: string;
  status: string;
  createdAt: string;
}

export interface Store {
  storeId: string;
  companyId: string;
  regionId: string;
  storeCode: string;
  storeName: string;
  storeType: string;
  openDate?: string;
  closeDate?: string;
  status: string;
  timezone: string;
  createdAt: string;
}

export interface Employee {
  employeeId: string;
  companyId: string;
  externalEmployeeRef?: string;
  firstName: string;
  lastName: string;
  hireDate: string;
  terminationDate?: string;
  employmentStatus: EmploymentStatus;
  employmentType: string;
  createdAt: string;
}

export interface EmployeeAssignmentHistory {
  assignmentId: string;
  employeeId: string;
  storeId: string;
  regionId: string;
  positionId: string;
  managerEmployeeId?: string;
  startDate: string;
  endDate?: string;
  isPrimaryAssignment: boolean;
  fteRatio: number;
  assignmentStatus: AssignmentStatus;
  createdAt: string;
}

export interface UserRoleAssignment {
  userRoleAssignmentId: string;
  userId: string;
  roleId: string;
  scopeType: ScopeType;
  companyId?: string;
  regionId?: string;
  storeId?: string;
  startAt: string;
  endAt?: string;
}

export interface ChecklistInstance {
  checklistInstanceId: string;
  checklistTemplateId: string;
  storeId: string;
  assignedEmployeeId?: string;
  auditorEmployeeId?: string;
  plannedAt?: string;
  startedAt?: string;
  completedAt?: string;
  status: ChecklistStatus;
  totalScore?: number;
  complianceRate?: number;
  createdAt: string;
}

export interface KpiActual {
  kpiActualId: string;
  kpiId: string;
  scopeType: ScopeType;
  companyId?: string;
  regionId?: string;
  storeId?: string;
  employeeId?: string;
  periodType: string;
  periodStart: string;
  periodEnd: string;
  actualValue: number;
  calculatedAt: string;
  sourceType: string;
}

export interface WorkforceNormPlan {
  normPlanId: string;
  companyId: string;
  regionId?: string;
  storeId?: string;
  positionId: string;
  periodStart: string;
  periodEnd: string;
  plannedHeadcount: number;
  plannedFte: number;
  approvedBy?: string;
  approvedAt?: string;
}

export interface SnapshotRun {
  snapshotRunId: string;
  snapshotDate: string;
  snapshotType: string;
  periodStart: string;
  periodEnd: string;
  generatedAt: string;
  generatedBy: string;
  sourceBatchNo?: string;
}

export interface IntegrationImportBatch {
  importBatchId: string;
  integrationSourceId: string;
  entityType: string;
  startedAt: string;
  finishedAt?: string;
  status: "pending" | "processing" | "completed" | "failed";
  rawFileName?: string;
  recordCount: number;
  errorCount: number;
}

export interface AuditEventLog {
  eventLogId: string;
  occurredAt: string;
  actorUserId?: string;
  eventType: string;
  entityName: string;
  entityId?: string;
  scopeType: ScopeType;
  companyId?: string;
  regionId?: string;
  storeId?: string;
  requestId?: string;
  ipAddress?: string;
  metadataJson: Record<string, unknown>;
}
